src/main/java/net/snowflake/ingest/streaming/internal/ArrowRowBuffer.java

```diff
@@ -945,7 +945,12 @@ private float convertRowToArrow(
               break;
             }
           case BINARY:
-            byte[] bytes = DataValidationUtil.validateAndParseBinary(value);
+            String maxLengthString = field.getMetadata().get(COLUMN_BYTE_LENGTH);
```

Comment:
@sfc-gh-tzhang Can you confirm that the `COLUMN_BYTE_LENGTH` is what we expect here?

Reply:
@sfc-gh-mnaides i'm not, based on my testing, byteLength and length always give me the same thing. I assume for char we should use length and for binary for we should use byte length?

Reply:
verified at the server side that length and byte length are the same value so we should we be good